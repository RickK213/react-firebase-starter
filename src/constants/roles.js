export const ROLES = {
  ADMIN: 'ADMIN',
  BASE_USER: 'BASE_USER'
};
