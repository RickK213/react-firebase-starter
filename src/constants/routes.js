export const ROUTES = {
  ADMIN: { label: 'Admin', path: '/admin' },
  ADMIN_DETAILS: { label: null, path: '/admin/:id' },
  ACCOUNT: { label: 'Account', path: '/account' },
  HOME: { label: 'Home', path: '/' },
  PASSWORD_FORGET: {
    label: 'Forgot Password?',
    path: '/account/password-forget'
  },
  SIGN_IN: { label: 'Sign In', path: '/account/sign-in' },
  SIGN_UP: { label: 'Sign Up', path: '/account/sign-up' }
};
