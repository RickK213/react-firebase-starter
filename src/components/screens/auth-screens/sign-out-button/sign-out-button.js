import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import { withFirebase } from '../../../firebase';
import { ROUTES } from '../../../../constants/routes';
import { buttonStyle } from '../auth-screen-styles';

export class SignOutButtonComponent extends Component {
  static propTypes = {
    firebase: PropTypes.object,
    history: PropTypes.object
  };

  static defaultProps = {
    firebase: null,
    history: null
  };

  constructor(props) {
    super(props);
    this.handleOnSignOutClick = this.handleOnSignOutClick.bind(this);
  }

  handleOnSignOutClick() {
    const { firebase, history } = this.props;
    firebase.doSignOut().then(() => {
      history.push(ROUTES.HOME.path);
    });
  }

  render() {
    return (
      <button
        type="button"
        onClick={this.handleOnSignOutClick}
        style={buttonStyle}
      >
        Sign Out
      </button>
    );
  }
}

export const SignOutButton = withRouter(withFirebase(SignOutButtonComponent));
