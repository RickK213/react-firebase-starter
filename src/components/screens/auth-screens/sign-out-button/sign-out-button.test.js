/* global React, create, shallow */
import { SignOutButtonComponent } from './sign-out-button';
import { ROUTES } from '../../../../constants/routes';

describe('<SignOutButtonComponent />', () => {
  it('should render correctly', () => {
    const component = create(<SignOutButtonComponent />).toJSON();
    expect(component).toMatchSnapshot();
  });

  describe('this.handleOnSignOutClick', () => {
    it('should call history.push if firebase call succeeds', async () => {
      const push = jest.fn();
      const history = { push };
      const doSignOut = jest.fn().mockResolvedValue('default');
      const firebase = { doSignOut };
      const component = shallow(
        <SignOutButtonComponent firebase={firebase} history={history} />
      );
      await component.instance().handleOnSignOutClick();
      expect(push).toHaveBeenCalledWith(ROUTES.SIGN_IN.path);
    });
  });
});
