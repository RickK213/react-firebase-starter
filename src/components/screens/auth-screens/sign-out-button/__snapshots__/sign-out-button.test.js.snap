// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignOutButtonComponent /> should render correctly 1`] = `
<button
  onClick={[Function]}
  style={
    Object {
      "padding": ".5rem",
      "textTransform": "uppercase",
    }
  }
  type="button"
>
  Sign Out
</button>
`;
